uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java

```diff
@@ -1266,13 +1266,26 @@ private boolean deleteFS(final FileSystem fileSystem) {
             FileUtils.delete(gitDir,
                              FileUtils.RECURSIVE | FileUtils.RETRY);
             fsManager.remove(fileSystem.getName());
+            cleanupSpaces(gitDir);
             return true;
         } catch (java.io.IOException e) {
             throw new IOException("Failed to remove the git repository.",
                                   e);
         }
     }
 
+    private void cleanupSpaces(File gitDir) throws java.io.IOException {
```

Comment:
I think it would be better to replace the word `space` for `parent[Dir]` as appropiate, to keep this code generic.

Reply:
+1 for @paulovmr - in this level of abstraction, we want more generic solution. this shouldn't know about spaces directly